assets/src/edit-story/app/media/useUploadMedia/useUploadMedia.js

```diff
@@ -62,6 +62,8 @@ function useUploadMedia({ media, pagingNum, mediaType, fetchMedia, setMedia }) {
             localResource: await getResourceFromLocalFile(file),
             file,
           }))
+        ).then((newLocalFiles) =>
```

Comment:
Can you use await and then together like this?

Reply:
https://codesandbox.io/s/nostalgic-meitner-6g4e3?file=/src/index.js

I tried something similar here to show what happens.

Then only gets run after the promise gets fulfilled.  The promise will only get fulfilled after the getResourceFromLocalFile gets resolved.

See here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then

Reply:
It works, although it could just as well be:

```js
localFiles = await Promise.all(
  files.reverse().map(async (file) => ({
    localResource: await getResourceFromLocalFile(file),
    file,
  }))
);
localFiles.filter((f) => f.localResource != null)
```

Reply:
So the actual code for that will be

`localFiles = localFiles.filter((...))` since filter actually returns the filtered array and doesn't change the actual array itself.

I personally prefered using "then" as it shows that its tied to the promise. However if you feel strongly about this, I'm happy to change it.

Reply:
I think it is confusing to have an await and a then. In my mind, it is one of the other. I don't think either way, just pick one.

Reply:
Chose the localFiles = localFiles.filter version.